File: refactoring-practice-1/src/main/typescript/extract_class/person.ts
/*
 * Emeljük ki a telefonszámot egy külön osztályba!
 */
export class Person {

    private officeAreaCode: string;
    private officeNumber: string;
    private name: string;

    public getOfficeTelephoneNumber(): string {
        return "(" + this.officeAreaCode + ") " + this.officeNumber;
    }

    public getOfficeAreaCode(): string {
        return this.officeAreaCode;
    }

    public setOfficeAreaCode(officeAreaCode: string) {
        this.officeAreaCode = officeAreaCode;
    }

    public getOfficeNumber(): string {
        return this.officeNumber;
    }

    public setOfficeNumber(officeNumber: string) {
        this.officeNumber = officeNumber;
    }

    public getName(): string {
        return this.name;
    }

    public setName(name: string) {
        this.name = name;
    }
}

File: refactoring-practice-1/src/main/java/dojo/extract_class/Person.java
package dojo.extract_class;

/*
 * Emeljük ki a telefonszámot egy külön osztályba!
 * Előtte írjuk meg a szükséges teszteket getOfficeAreaCode, getOfficeNumber metódusok számára!
 */
public class Person {

	private String officeAreaCode;
	private String officeNumber;
	private String name;
	
	public String getOfficeTelephoneNumber() {
		return "(" + officeAreaCode + ") " + officeNumber;
	}
	
	public String getOfficeAreaCode() {
		return officeAreaCode;
	}
	
	public void setOfficeAreaCode(String officeAreaCode) {
		this.officeAreaCode = officeAreaCode;
	}
	
	public String getOfficeNumber() {
		return officeNumber;
	}
	
	public void setOfficeNumber(String officeNumber) {
		this.officeNumber = officeNumber;
	}
	
	public String getName() {
		return name;
	}
	
	public void setName(String name) {
		this.name = name;
	}
}
